### PR TITLE
neutron: run multiple api workers

### PR DESCRIPTION
**Pull request by Itxaka**

### CODE REVIEW

**Comment by Itxaka** [comment]
[(2) neutron: Run multiple api_workers](https://trello.com/c/gX8NK4yn/402-2-neutron-run-multiple-api-workers)

File: chef/cookbooks/neutron/templates/default/neutron.conf.erb [hunk 1]
```diff
@@ -204,7 +204,7 @@ use_ssl = <%= @ssl_enabled ? "True" : "False" %>
 # default is equal to the number of CPUs available for best performance.
 # (integer value)
 #api_workers = <None>
-api_workers = 0
+api_workers = <%= [node["cpu"]["total"], 2, 4].sort[1] %>
```
Comment (Itxaka) [inline chef/cookbooks/neutron/templates/default/neutron.conf.erb]: Im still not sure the reasoning behind this.

Anything lower than 4 cpus will use 2 workers.
Anything over 6 cpus will use 4 workers always.

It does not seem to increase workers linearly with cpu, which is something that I would expect. 

Maybe it needs a more complex helper to calculate the amount of workers properly?

Comment (toabctl) [inline chef/cookbooks/neutron/templates/default/neutron.conf.erb]: @Itxaka why not leaving the default (None)?
```
# Number of separate API worker processes for service. If not specified, the
# default is equal to the number of CPUs available for best performance.
# (integer value)
```

Comment (Itxaka) [inline chef/cookbooks/neutron/templates/default/neutron.conf.erb]: Yeah, I guess thats the correct way. I thougth that the method used across all cookbooks for calculating the api_workers was due to not have workers on all cpus, but it doesnt look like it. Im not sure whats the real policy here regarding workers == cpus or we want to have less workers or whatnot. But I guess thats out of scope for this :D

Comment (dirkmueller) [inline chef/cookbooks/neutron/templates/default/neutron.conf.erb]: @Itxaka we definitely don't want to have workers==cpu as that is excessive on big machines (think 32 cores and more). also at least in the past neutron used to have terrible races with large number of workers.. better be conservative and use the same code like the others, even if its bad.

File: chef/cookbooks/neutron/templates/default/neutron.conf.erb [hunk 2]
```diff
@@ -204,7 +204,6 @@ use_ssl = <%= @ssl_enabled ? "True" : "False" %>
 # default is equal to the number of CPUs available for best performance.
 # (integer value)
 #api_workers = <None>
-api_workers = 0
```
Comment (dirkmueller) [inline chef/cookbooks/neutron/templates/default/neutron.conf.erb]: I suggest to not just use the default as that is excessive on large core-machines (think 32 cores and more). also at least in the past neutron used to have terrible races with api_workers !=0. we might want to be conservative here and use exactly the same code like everywhere else (```<%= [node["cpu"]["total"], 2, 4].sort[1] %>```)

File: chef/cookbooks/neutron/templates/default/neutron.conf.erb (hunk 1 above)
Comment (toabctl) [inline chef/cookbooks/neutron/templates/default/neutron.conf.erb]: @Itxaka then let's go back to your initial proposal.

File: chef/cookbooks/neutron/templates/default/neutron.conf.erb (hunk 2 above)
Comment (Itxaka) [inline chef/cookbooks/neutron/templates/default/neutron.conf.erb]: Updated!

**Comment by toabctl** [comment]
Tempest (linuxbridge) passed. It failed because of radosgw. merging...

**Pull request merged**